posts/2021/08/github-actions-benchmarks.md

```diff
@@ -0,0 +1,336 @@
+to reduce that time without sacrificing sensitivity? We need to remember that `asv` is running several
+passes and repeats to reduce the measurement error, but maybe some of those default counter-measures
+are not needed. Namely:
+
+- The benchmark runs with `--interleave-processes`, but it can be disabled with `--no-interleave-processes`.
+The help message for this flag says:
+
+    > Interleave benchmarks with multiple processes across commits. This can avoid measurement biases
+    > from commit ordering, can take longer.
+
+    How much longer? Does it help keep error under control? We should measure that.
+
+- By default, all tests are run several times, with different schedules. There are two
+[benchmark attributes](https://asv.readthedocs.io/en/stable/benchmarks.html#timing-benchmarks)
+that govern these settings: `processes` and `repeat`. `processes` defaults to `2`, which means that
+the full suite will be run twice per commit. If we only do one pass (`processes=1`), we will reduce
+the running time in half, but will we lose too much accuracy?
+
+To answer both questions, we added more entries to the data collection workflow shown above by
+[parameterizing the `asv` command-line options](https://github.com/jaimergp/scikit-image/blob/e561996/.github/workflows/benchmarks-cron.yml#L14-L23).
+
+These are the results!
+
+## No process interleaving
+
+Disabling process interleaving should be faster and maybe the accuracy loss is not that bad. But...
+how bad? Here are the results:
+
+![Reliability of benchmarks in GitHub Actions, no interleaving](/images/2021/08/github-actions-benchmark-no-interleaving.png)
+
+* Average time taken: 1h39min
+* Minimum and maximum ratios observed: 0.43, 1.5
+* Mean and standard deviation of ratios: 0.99, 0.07
+* Proportion of false positives: 6/66 = 9.99%
+
+## Single-pass with `processes=1`
+
+In this configuration, we expect a drastic 50% running time reduction, since we will only do one pass
+per commit, instead of two. However, the accuracy loss might be too dramatic... Let's see!
+
+![Reliability of benchmarks in GitHub Actions, single-process](/images/2021/08/github-actions-benchmark-single-process.png)
+
+* Average time taken: 1h7min
+* Minimum and maximum ratios observed: 0.51, 2.76
+* Mean and standard deviation of ratios: 1.01, 0.07
+* Proportion of false positives: 8/64 = 12.5%
+
+At first sight, those clouds look very spread! The number of false positives is also larger.
+
+## Summary of the strategies
+
+Let's take a look at the three strategies now. The main columns are **runtime** and **%FP**
+(percentage of false positives). We want the smallest %FP at the shortest runtime.
+
+| Strategy        | Runtime  &nbsp;| %FP  | Min  | Max  | Mean &nbsp;| Std  |
+|-----------------|---------|-------|------|------|------|------|
+| Default         | 1h55    | 3.7%  | 0.51 &nbsp;| 1.36 | 1.00 | 0.05 |
+| No interleaving &nbsp;| 1h39    | 9.99% &nbsp;| 0.43 | 1.50 | 0.99 | 0.07 |
```

Comment:
```suggestion
| No interleaving &nbsp;| 1h39    | 9.99 &nbsp;| 0.43 | 1.50 | 0.99 | 0.07 |
```

(The other rows don't have % here and it is stated in the header row)

Reply:
I think they do have the `%`, but yes, we can remove it if it's in the header!